app/src/main/java/com/example/tasky/MainActivity.kt

```diff
@@ -1,47 +1,42 @@
 package com.example.tasky
 
 import android.os.Bundle
+import android.util.Log
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.Scaffold
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.activity.viewModels
+import androidx.compose.runtime.mutableStateOf
+import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
+import com.example.tasky.feature_login.presentation.LoginScreenContent
+import com.example.tasky.feature_login.presentation.RegisterAccountContent
+import com.example.tasky.feature_splash.domain.SplashViewModel
 import com.example.tasky.ui.theme.TaskyTheme
 
 class MainActivity : ComponentActivity() {
+
+    private val splashViewModel: SplashViewModel by viewModels()
+    private val isLoggedInState = mutableStateOf(false)
+
+    companion object {
+        private const val TAG = "MainActivity"
+    }
     override fun onCreate(savedInstanceState: Bundle?) {
+        Log.d(TAG, "onCreate is called")
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
+        splashScreen.setKeepOnScreenCondition {
+            //TODO add logic to check navigation
+            isLoggedInState.value = splashViewModel.checkIfUserIsLoggedIn()
```

Comment:
`checkIfUserLoggedIn()` sounds more like a function that purely does something rather than a function that has a return value and delivers a result. Could something like `isUserLoggedIn()` be a better pick here which hints more towards returning a boolean?